packages/database/src/migrations/20210610111044-AddReportForMosquitoSpecieesMapOverlay-modifies-data.js

```diff
@@ -0,0 +1,82 @@
+'use strict';
+
+import { generateId, insertObject } from '../utilities';
+
+var dbm;
+var type;
+var seed;
+
+/**
+ * We receive the dbmigrate dependency from dbmigrate initially.
+ * This enables us to not have to rely on NODE_PATH.
+ */
+exports.setup = function (options, seedLink) {
+  dbm = options.dbmigrate;
+  type = dbm.dataType;
+  seed = seedLink;
+};
+
+const permissionGroupNameToId = async (db, name) => {
+  const record = await db.runSql(`SELECT id FROM permission_group WHERE name = '${name}'`);
+  return record.rows[0] && record.rows[0].id;
+};
+
+const REPORT = {
+  id: generateId(),
+  code: 'PG_Strive_Mosquito_Species_MapOverlay',
+  config: {
+    fetch: {
+      aggregations: [
+        {
+          type: 'RAW',
+          config: {
+            dataSourceEntityType: 'facility',
+          },
+        },
+      ],
+      dataElements: ['STRVEC_AE-AT13'],
+    },
+    transform: [
+      {
+        transform: 'select',
+        "'numerator'": '1',
+        "'organisationUnitCode'": '$row.organisationUnit',
+        "'name'": '$row.value',
+      },
+      {
+        transform: 'select',
+        "'denominator'":
+          'sum($where(f($otherRow) = equalText($otherRow.organisationUnitCode, $row.organisationUnitCode)).numerator)',
```

Comment:
Having thought this one over a bit, I rekon using the `$where` is best for now 👍 

I do like your idea of having some a pre-defined special `$where` functions, but perhaps let's hold off a little bit til this functionality gets used more often and we know how best to generalise? I think this is the first report ever that uses the `$where` 🎉

Reply:
Ha, `$where` is a good stuff!